tape/src/main/java/com/squareup/tape2/QueueFile.java

```diff
@@ -499,6 +496,10 @@ private void expandIfNecessary(long dataLength) throws IOException {
     }
 
     fileLength = newLength;
+
+    if (zero && count != 0) {
```

Comment:
I will be zero when the data is contiguous (not wrapped) before the expansion.

But you are right, we don't need to check it -- ringErase handles it.